server/runner/schemasync/syncer.go

```diff
@@ -449,3 +444,49 @@ func syncDBSchema(ctx context.Context, store *store.Store, database *api.Databas
 	}
 	return nil
 }
+
+func convertDBSchema(schema *db.Schema) *storepb.DatabaseMetadata {
+	// TODO(d): finish the implementation.
+	// TODO(d): add unit test.
+	databaseMetadata := &storepb.DatabaseMetadata{
+		Name:         schema.Name,
+		CharacterSet: schema.CharacterSet,
+		Collation:    schema.Collation,
+	}
+
+	schemaNameMap := make(map[string]bool)
+	schemaTableMap := make(map[string][]db.Table)
+	for _, table := range schema.TableList {
+		schemaNameMap[table.Schema] = true
+		schemaTableMap[table.Schema] = append(schemaTableMap[table.Schema], table)
+	}
+	var schemaNames []string
+	for schemaName := range schemaNameMap {
+		schemaNames = append(schemaNames, schemaName)
+	}
```

Comment:
`schemaNameMap` seems to be unnecessary here.

Reply:
It's by design that we're going to introduce view map and we need a superset of all them.